orion/controllers/programs/programsController.go

```diff
@@ -81,6 +79,40 @@ func DeleteProgram(c *gin.Context) {
   return
 }
 
-func CheckValidProgram() bool {
+func CheckValidProgram(c *gin.Context) bool {
```

Comment:
nice try!
`gin` is an external package for HTTP request and response handling. We'll talk more about that later.

Instead, the input here should be a Program. We are going to use function CheckValidProgram() to check a passed in Program. If it's valid we return true, otherwise we return false.